lang/cs/Org.Apache.REEF.IMRU/OnREEF/Driver/ServiceAndContextConfigurationProvider.cs

```diff
@@ -70,13 +89,13 @@ internal ServiceAndContextConfigurationProvider(IPartitionedInputDataSet dataset
         /// <returns>Whether failed evaluator is master or not</returns>
         internal void RemoveEvaluatorIdFromPartitionIdProvider(string evaluatorId)
         {
-            if (!_partitionIdProvider.ContainsKey(evaluatorId))
+            if (!_partitionContextIdProvider.ContainsKey(evaluatorId))
```

Comment:
I am talking about the entire method. You make three trips to the same entry in the Dictionary: `_partitionContextIdProvider.ContainsKey(evaluatorId)`, then `_partitionContextIdProvider[evaluatorId]`, and, finally, `_partitionContextIdProvider.Remove(evaluatorId)`. My point is that you can do it with fewer calls, e.g.
```C#
PartitionDescriptorContextIdBundle partition;
if(_partitionContextIdProvider.TryGetValue(evaluatorId, out partition))
{
    _availablePartitionDescriptorContextIds.Push(partition);
    _partitionContextIdProvider.Remove(evaluatorId);
}
// ...
```
i.e. at least there is no `[]`. (I wish C# `.Remove()` returned the vaule, and not bool)